criu/include/kerndat.h

```diff
@@ -82,6 +85,15 @@ struct kerndat_s {
 	bool has_openat2;
 	bool has_rseq;
 	bool has_ptrace_get_rseq_conf;
+	/*
+	 * For non-root CRIU we use uname(2) to remember the nodename and
+	 * release. If the kerndat file has a different nodename or release
+	 * we assume that there has been a reboot and we need to recreate it.
+	 * nodename in addition to release in case the kerndat file is on a
+	 * NFS shared file system.
+	 */
+	char release[RELEASE_SIZE];
+	char nodename[NODENAME_SIZE];
```

Comment:
This can probably be removed for the new, XDG_RUNTIME_DIR based kdat approach.

Reply:
Done. In non-root mode if `XDG_RUNTIME_DIR` is defined we'll use it to form the cache filename and try to load/store to that file; if undefined we'll init the kdat from scratch and discard on exit.